thewalrus/fock_gradients.py

```diff
@@ -355,3 +357,73 @@ def grad_beamsplitter(T, theta, phi):  # pragma: no cover
                     )
 
     return grad_theta, grad_phi
+
+@lru_cache()
+def partition(num_modes: int, n_current: int, cutoff: int)-> Tuple[Tuple[int, ...], ...]:
```

Comment:
note that in the rest of the files we don't do hinting.

Reply:
I'm sorry Nico I haven't got your point, what is the hint here you are talking about

Reply:
I think Nico is referring to the type hints here (e.g., `cutoff: int`).

In the rest of The Walrus we don't do type hinting, however I don't mind too much  if it is used in some places. @nquesada what is your preference?

Reply:
Thanks, josh, I got the idea. If we don't use that overall in SF and TW, I could delete them.

Reply:
What if instead we slowly type-hint the rest? It can't be bad to have additional information...

Reply:
I have to say that I am not a big fan of this option.